Index: src/ru/javawebinar/basejava/storage/Storage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/ru/javawebinar/basejava/storage/Storage.java	(date 1560866798000)
+++ src/ru/javawebinar/basejava/storage/Storage.java	(date 1560866798000)
@@ -0,0 +1,20 @@
+package ru.javawebinar.basejava.storage;
+
+import ru.javawebinar.basejava.model.Resume;
+
+public interface Storage {
+
+    void clear();
+
+    void update(Resume r);
+
+    void save(Resume r);
+
+    Resume get(String uuid);
+
+    void delete(String uuid);
+
+    Resume[] getAll();
+
+    int size();
+}
Index: src/ru/javawebinar/basejava/storage/ArrayStorage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/ru/javawebinar/basejava/storage/ArrayStorage.java	(revision d9b94d5a49760f1968a56862b493c63086bf4c2d)
+++ src/ru/javawebinar/basejava/storage/ArrayStorage.java	(date 1560866798000)
@@ -7,7 +7,7 @@
 /**
  * Array based storage for Resumes
  */
-public class ArrayStorage {
+public class ArrayStorage implements Storage{
 
     private static final int STORAGE_LIMIT = 10000;
 
